server/debug/src/call.rs

```diff
@@ -0,0 +1,160 @@
+        let (calling_convention, return_type, arg_types) = match function_type {
+            Type::Function { calling_convention, type_index, args } => {
+                (calling_convention, type_index, args)
+            }
+            _ => return Err(io::Error::new(io::ErrorKind::Other, "cannot call a non-function")),
+        };
+
+        if calling_convention != 0 { // CV_CALL_NEAR_C
+            return Err(io::Error::new(io::ErrorKind::Other, "unsupported calling convention"));
+        }
+
+        let mut args = Iterator::zip(args.iter(), arg_types.into_iter());
+        let mut new_context = context.clone();
+
+        let return_type = symbols.type_from_index(module, return_type)?;
+        let return_size = return_type.size(symbols, module);
+        if return_size > 8 {
+            let stack_pointer = new_context.stack_pointer() - return_size;
+            new_context.set_stack_pointer(stack_pointer);
+
+            let context = new_context.as_raw_mut();
+            context.Rcx = stack_pointer as winapi::DWORD64;
+        } else if let Some((arg, arg_type)) = args.next() {
+            let arg_type = symbols.type_from_index(module, arg_type)?;
+            let (value, float) = write_value(&arg, &arg_type, child, &mut new_context)?;
+
+            let context = new_context.as_raw_mut();
+            if !float {
+                context.Rcx = value as winapi::DWORD64;
+            } else {
+                context.FltSave.XmmRegisters[0].Low = value as winapi::ULONGLONG;
+            }
+        }
+
+        if let Some((arg, arg_type)) = args.next() {
+            let arg_type = symbols.type_from_index(module, arg_type)?;
+            let (value, float) = write_value(&arg, &arg_type, child, &mut new_context)?;
+
+            let context = new_context.as_raw_mut();
+            if !float {
+                context.Rdx = value as winapi::DWORD64;
+            } else {
+                context.FltSave.XmmRegisters[1].Low = value as winapi::ULONGLONG;
+            }
+        }
+
+        if let Some((arg, arg_type)) = args.next() {
+            let arg_type = symbols.type_from_index(module, arg_type)?;
+            let (value, float) = write_value(&arg, &arg_type, child, &mut new_context)?;
+
+            let context = new_context.as_raw_mut();
+            if !float {
+                context.R8 = value as winapi::DWORD64;
+            } else {
+                context.FltSave.XmmRegisters[2].Low = value as winapi::ULONGLONG;
+            }
+        }
+
+        if let Some((arg, arg_type)) = args.next() {
+            let arg_type = symbols.type_from_index(module, arg_type)?;
+            let (value, float) = write_value(&arg, &arg_type, child, &mut new_context)?;
+
+            let context = new_context.as_raw_mut();
+            if !float {
+                context.R9 = value as winapi::DWORD64;
+            } else {
+                context.FltSave.XmmRegisters[3].Low = value as winapi::ULONGLONG;
+            }
+        }
+
```

Comment:
Yes- in the Win64 calling convention, the first four arguments are passed in registers RCX, RDX, R8, and R9. Further arguments are passed on the stack.